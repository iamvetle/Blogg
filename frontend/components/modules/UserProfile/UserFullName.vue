<template>
    <h2 data-test="user-full-name" class="text-4xl leading-[52px] font-medium tracking-[-0.03em]">
        {{ first_name }} {{ last_name }}
    </h2>
</template>

<script setup lang="ts">
/**
 * Is used in the /user/[id] pages to render the full name of the user
 */

defineProps<{
    first_name: string;
    last_name: string;
}>();

</script>

<style scoped></style>