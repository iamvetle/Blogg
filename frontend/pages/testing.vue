<template>
  <div class="flex">

    <form @submit.prevent method="post" class="flex flex-col" enctype="multipart/form-data">

      <label for="name">Name:</label>
      <input type="text" name="name" id="name">

      <label for="email">Email</label>
      <input type="email" name="email" id="email">

      <label for="message">Message:</label>
      <textarea name="message" id="" cols="30" rows="10" placeholder="Write a message here"></textarea>

      <label for="file">File:</label>
      <input type="file" name="file" id="file">

      <button type="submit">Submit</button>
    </form>

  </div>
</template>

<script setup lang="ts">
onMounted(() => {

  const form = document.querySelector("form")

  form?.addEventListener("submit", (e: any) => {


    const formData = new FormData(form)

    for (let item of formData) {
      console.log(item[0], item[1])
    }
    
    // ta axios of prÃ¸v fetch her
  })





})

// watchEffect(() => {

// })

</script>

<style scoped></style>