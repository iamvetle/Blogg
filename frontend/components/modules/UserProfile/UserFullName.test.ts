import UserFullName from './UserFullName.vue';
import { shallowMount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';


let wrapper: any;
let pinia: any = createTestingPinia();

// let generalStore; 
// let postStore; 
// let loggedInUserStore; 
// let paginationStore; 

let firstNameProp = "test_first_name"
let lastNameProp = "test_last_name"

const factory = () => {
    return shallowMount(UserFullName, {
        global: {
            plugins: [pinia],
            components: {},
            mocks: {},
            stubs: {
                Name: true
            },
        },
        props: {
            first_name: firstNameProp,
            last_name: lastNameProp
        },
        slots: {}
    })
};

describe('Testing the userfullname component that is used inside of the id user page', () => {

    beforeEach(() => {
    });

    afterEach(() => {
        if (wrapper) {
            wrapper.unmount();
        }
    });
    test('Should exist', () => {
        wrapper = factory()

        expect(wrapper.exists()).toBe(true)
    })
    test('Should have h2 element', () => {
        wrapper = factory()

        expect(wrapper.find("h2").exists()).toBe(true)
    })
    test('Should render the props', () => {
        wrapper = factory()

        const firstProp = wrapper.props("first_name")
        const lastProp = wrapper.props("last_name")

        expect(wrapper.text()).toContain(firstProp)
        expect(wrapper.text()).toContain(lastProp)
    })
    test('Should render the entire name', () => {
        wrapper = factory()

        const firstProp = wrapper.props("first_name")
        const lastProp = wrapper.props("last_name")

        const targetElement = wrapper.find("[data-test='user-full-name']")
        expect(targetElement.text()).toBe(`${firstProp} ${lastProp}`)
    })
    test('Should match snapshot', () => {
        wrapper = factory()

        expect(wrapper.html()).toMatchSnapshot()
    })

});