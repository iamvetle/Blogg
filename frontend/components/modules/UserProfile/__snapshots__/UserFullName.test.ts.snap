// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the userfullname component that is used inside of the id user page > Should match snapshot 1`] = `"<h2 data-test=\\"user-full-name\\" class=\\"text-4xl leading-[52px] font-medium tracking-[-0.03em]\\">test_first_name test_last_name</h2>"`;
